cea/optimization/master/summarize_network.py

```diff
@@ -27,52 +26,35 @@
 __status__ = "Production"
 
 
-def network_main(locator, total_demand, building_names, config, gv, key):
+def network_main(locator, buildings_in_this_network, ground_temp, num_tot_buildings, network_type, key, network_barcode=""):
     """
     This function summarizes the distribution demands and will give them as:
     - absolute values (design values = extreme values)
     - hourly operation scheme of input/output of distribution
 
     :param locator: locator class
     :param total_demand: dataframe with total demand of buildings
-    :param building_names: vector with names of buildings
-    :param gv: global variables class
+    :param buildings_in_this_network: vector with names of buildings
     :param key: when called by the optimization, a key will provide an id for the individual
         and the generation.
     :type locator: class
     :type total_demand: list
-    :type building_names: vector
-    :type gv: class
+    :type buildings_in_this_network: vector
     :type key: int
     :return: csv file stored in locator.get_optimization_network_results_folder() as fName_result
         where fName_result: FIXME: what?
     :rtype: Nonetype
     """
 
+    # local variables
     t0 = time.clock()
-    weather_data = epwreader.epw_reader(config.weather)[['year', 'drybulb_C', 'wetbulb_C','relhum_percent',
-                                                              'windspd_ms', 'skytemp_C']]
-    ground_temp = calc_ground_temperature(locator, config, weather_data['drybulb_C'], depth_m=10)
-    # import properties of distribution
-    network_type = config.thermal_network.network_type
-    list_network_name = ['', '']  # config.thermal_network.network_names
-    num_buildings_network = total_demand.Name.count()
-    if len(list_network_name) == 0:
-        network_name = ''
-        pipes_tot_length = pd.read_csv(locator.get_thermal_network_edge_list_file(network_type, network_name),
-                                       usecols=['pipe length']).sum().values
-    else:
-        for i, network_name in enumerate(list_network_name):
-            if i == 0:
-                pipes_tot_length = pd.read_csv(
-                    locator.get_thermal_network_edge_list_file(network_type, network_name),
-                    usecols=['pipe length']).sum().values
-            else:
-                pipes_tot_length = pipes_tot_length + pd.read_csv(
-                    locator.get_thermal_network_edge_list_file(network_type, network_name),
-                    usecols=['pipe length']).sum().values
-
-    ntwk_length = pipes_tot_length.sum() * num_buildings_network / gv.num_tot_buildings
+    num_buildings_network = len(buildings_in_this_network)
+    date = pd.read_csv(locator.get_demand_results_file(buildings_in_this_network[0])).DATE.values
+
+    # CALCULATE RELATIVE LENGTH OF THIS NETWORK
```

Comment:
this is an _obvious_ candidate for "extract method" refactoring. there are many like this. once we get the code to work, we should go to town on this.

```
# also, no need to call it ntwk when network is so much easier to read
network_length = calculate_relative_length(network_type, num_buildings_network, num_tot_buildings)
```

Reply:
yep...it would be good to create issues about this. It will not be part of the refactoring of this PR